packages/build/src/packaging/package/tarball.spec.ts

```diff
@@ -1,12 +1,38 @@
+import { expect } from 'chai';
+import { spawnSync } from 'child_process';
 import { promises as fs } from 'fs';
+import path from 'path';
+import rimraf from 'rimraf';
+import { promisify } from 'util';
 import { withTempPackageEach } from '../../../test/helpers';
 import { createPackage } from './create-package';
 
 describe('package tarball', () => {
   const tmpPkg = withTempPackageEach();
 
-  it('packages the executable(s)', async() => {
-    const tarball = await createPackage(tmpPkg.tarballDir, 'linux-x64', tmpPkg.pkgConfig);
-    await fs.access(tarball.path);
+  describe('packages the executable(s)', async() => {
+    const tmpDir = path.join(__dirname, '..', '..', '..', 'tmp', `pkg-${Date.now()}-${Math.random()}`);
+
+    beforeEach(async() => {
+      await fs.mkdir(tmpDir, { recursive: true });
+    });
+
+    afterEach(async() => {
+      await promisify(rimraf)(tmpDir);
+    });
+
+    it('with correct structure', async() => {
+      const tarball = await createPackage(tmpPkg.tarballDir, 'linux-x64', tmpPkg.pkgConfig);
+      await fs.access(tarball.path);
+      const tarname = path.basename(tarball.path);
+
+      const unzip = spawnSync('tar', [
+        'xf', tarball.path, '--directory', tmpDir
+      ], { encoding: 'utf-8' });
```

Comment:
I think `tar tf` would also be fine here, we don’t need to unpack the tarball to see what’s inside